elastic/kibana
# Introduce a license check for the User Profile Suggest API.

## Summary

Introduce a license check for the User Profile Suggest API (available in `Standard+`).

### How to test

The idea is to first start Elasticsearch and Kibana with the trial license and use Kibana DevTools to see if `_suggest` endpoint works (it should). Then repeat the same with the basic license and make sure that `_suggest` endpoint throws:
```bash
# ES and Kibana with trial
yarn es snapshot --license trial
yarn start --plugin-path x-pack/test/security_api_integration/fixtures/user_profiles/user_profiles_consumer/

# ES and Kibana with basic
yarn es snapshot --license basic
yarn start --plugin-path x-pack/test/security_api_integration/fixtures/user_profiles/user_profiles_consumer/
```

```http
POST kbn:/internal/user_profiles_consumer/_suggest
{
  "name": "elastic"
}

## This might not fail even on basic since ES snapshot with
## the corresponding license check might not be available at
## the review time yet: https://github.com/elastic/elasticsearch/pull/89990
POST /_security/profile/_suggest
{
  "name": "elastic"
}

```

cc @jonathan-buttner @cnasikas 

__Fixes: https://github.com/elastic/kibana/issues/139430__
__Blocked by: ~~https://github.com/elastic/elasticsearch/pull/89990~~, ~~https://github.com/elastic/kibana/pull/140508~~__
